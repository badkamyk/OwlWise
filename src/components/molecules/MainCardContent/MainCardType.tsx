export type MainCardType = {
	title: string;
	description: string;
	src: string;
};
