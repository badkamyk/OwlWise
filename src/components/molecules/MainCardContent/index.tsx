import MainCardDescription from '../../atoms/MainCardDescription'
import MainCardImg from '../../atoms/MainCardImg'
import MainCardTitle from '../../atoms/MainCardTitle'
import { MainCardType } from './MainCardType'

export default function MainCardText({ title, description }: MainCardType) {
    return (
        <div className="relative rounded-[15px] bg-white p-6">
            <div className="space-y-4">
                <MainCardTitle text={title} />
                <MainCardDescription text={description} />
                <MainCardImg src="https://nuxt.com/assets/home/ux-fast-light.svg" alt="Image with particles" />
            </div>
        </div>
    )
}
