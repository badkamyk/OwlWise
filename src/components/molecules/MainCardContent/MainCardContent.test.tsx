import { render, screen } from "@testing-library/react";
import "intersection-observer";
import { expect, it } from "vitest";
import MainCardContent from "./index";

it("renders MainCardContent correctly", () => {
	const { container } = render(
		<MainCardContent
			title={"title"}
			description={"description"}
			src={"https://via.placeholder.com/150"}
		/>
	);
	expect(
		container.getElementsByClassName("relative rounded-[15px] bg-white p-6")
	).toBeTruthy();
	expect(screen.getByText("title")).toBeTruthy();
	expect(screen.getByText("description")).toBeTruthy();
	expect(screen.getByRole("img")).toBeTruthy();
	expect(screen.getByRole("img")).toHaveProperty(
		"src",
		"https://via.placeholder.com/150"
	);
	expect(container).toMatchSnapshot();
});
