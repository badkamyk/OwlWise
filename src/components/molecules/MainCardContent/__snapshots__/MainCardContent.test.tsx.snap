// Vitest Snapshot v1

exports[`renders MainCardContent correctly 1`] = `
<div>
  <div
    class="relative rounded-[15px] bg-white p-6"
    style="opacity: 0; transform: translateY(50px);"
  >
    <div
      class="space-y-4"
    >
      <img
        alt="Image with particles"
        src="https://via.placeholder.com/150"
      />
      <p
        class="text-lg font-semibold text-slate-800"
      >
        title
      </p>
      <p
        class="mb-4 text-gray-900 text-md"
        data-testid="main-content-paragraph"
      >
        description
      </p>
    </div>
  </div>
</div>
`;
